stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/internal/Linkify.kt

```diff
@@ -0,0 +1,212 @@
+package io.getstream.chat.android.ui.common.internal
+
+import android.annotation.SuppressLint
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.Spanned
+import android.text.method.LinkMovementMethod
+import android.text.style.URLSpan
+import android.text.util.Linkify
+import android.widget.TextView
+import androidx.core.util.PatternsCompat
+import java.util.ArrayList
+import java.util.Collections
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+/**
+ * Linkify links the part of the text based on matching pattern.
+ *
+ * This class is a simplified version of [Linkify] and differs only in one following way
+ * It doesn't remove any existing URLSpan from the Spannable.
+ */
+internal object Linkify {
+
+    private val COMPARATOR: Comparator<LinkSpec> = object : Comparator<LinkSpec> {
+        override fun compare(a: LinkSpec, b: LinkSpec): Int {
+            if (a.start < b.start) {
```

Comment:
You can use `when` instead of `if/else`: 
```
return when {
                a.start < b.start -> -1
                a.start > b.start, a.end < b.end -> 1
                else -> 0
            }
```